staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -246,6 +246,17 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 	}
 
 	<-stopCh
+
+	if s.GenericAPIServer.AuditBackend != nil {
+		// TODO: make timeout configurable
+		shutdownStopCh := make(chan struct{})
+		go func() {
+			defer close(shutdownStopCh)
+			<-time.After(time.Minute)
```

Comment:
I prefer leaving this out, for the exact reason @sttts mentioned: in most setups SIGKILL will follow anyway